tests/collections/llm/gpt/model/test_hyena.py

```diff
@@ -0,0 +1,685 @@
+                f"-F32R{evo2_config.fp32_residual_connection}"
+                f"-FCE{evo2_config.cross_entropy_loss_fusion}"
+                f"-AIC{not args.no_average_in_collective}"
+                f"-PEOD{args.eod_pad_in_loss_mask}"
+                f"-BO{args.add_bias_output}"
+                f"-GCLP{args.clip_grad}"
+                f"-HDO{args.hidden_dropout}"
+                f"-ADO{args.attention_dropout}"
+                f"-LR{args.lr}-MINLR{args.min_lr}-WUSTEPS{args.warmup_steps}-WD{args.wd}"
+                f"-GRFP32{args.grad_reduce_in_fp32}-FP8WG{args.fp8_wgrad and args.fp8}"
+                f"-OGR{args.overlap_grad_reduce}-OPG{args.overlap_param_gather}"
+                f"-NODES{args.num_nodes}-FP8{args.fp8}"
+            ),
+            group=args.wandb_group,
+            job_type=args.wandb_job_type,
+            id=args.wandb_run_id,  # set this to use the same curve name for restarts.
+            project=args.wandb_project,
+            save_dir=args.experiment_dir,
+        )
+        loggers.append(wandb_logger)
+        nemo_logger_kwargs["wandb"] = wandb_logger
+    tb_logger = TensorBoardLogger(
+        save_dir="dummy",  ## NOTE: this gets overwritten by default
+    )
+    nemo_logger_kwargs["tensorboard"] = tb_logger
+    loggers.append(tb_logger)
+
+    nemo_logger = NeMoLogger(log_dir=args.experiment_dir, **nemo_logger_kwargs)
+    ddp: DistributedDataParallelConfig = DistributedDataParallelConfig(
+        check_for_nan_in_grad=True,
+        overlap_grad_reduce=args.overlap_grad_reduce,
+        overlap_param_gather=args.overlap_param_gather,  # Verify that this works using
+        grad_reduce_in_fp32=args.grad_reduce_in_fp32,
+        align_param_gather=args.align_param_gather,
+        average_in_collective=not args.no_average_in_collective,
+    )
+    # Initialize Megatron Strategy and Trainer.
+    strategy = nl.MegatronStrategy(
+        ddp=ddp,
+        tensor_model_parallel_size=args.tensor_parallel_size,
+        pipeline_model_parallel_size=args.pipeline_model_parallel_size,
+        context_parallel_size=args.context_parallel_size,
+        pipeline_dtype=torch.bfloat16,
+        sequence_parallel=args.sequence_parallel,
+        ckpt_load_optimizer=True,
+        ckpt_save_optimizer=True,
+        ckpt_async_save=args.ckpt_async_save,
+        save_ckpt_format=args.ckpt_format,
+        ckpt_load_strictness="log_all",  # or rebasing to https://github.com/NVIDIA/NeMo/pull/11988/files#diff-7667eae242a8ef776bff78cd08e79bc81df4896a450f0a781f6ed317a3dfb7ffR139
+    )
+    trainer = nl.Trainer(
+        devices=args.devices,
+        num_nodes=args.num_nodes,
+        max_steps=args.max_steps,
+        accelerator="gpu",
+        strategy=strategy,
+        logger=loggers,
+        callbacks=callbacks,
+        log_every_n_steps=args.log_every_n_steps,
+        limit_val_batches=args.limit_val_batches,
+        num_sanity_val_steps=0,
+        use_distributed_sampler=True,
```

Comment:
@JRD971000 are you sure this should be True? I thought that this was to use the lightning distributed sampler, but we use a custom megatron sampler as part of our megatron strategy and data loaders. Did something change recently that requires us to set this to True?